### PR TITLE
fix: streaming error handling

### LINKED ISSUE #88: Bug in streaming client
Hey, we ran into a bug when using the streaming client where it gets a 401 back from the Looker server and ignores the error.

I suspect the following code in the `build_response` method of the streaming client:
https://github.com/looker-open-source/looker-sdk-ruby/blob/92cb759090d44f9fa8524a552508d8e6cbcdcb6e/lib/looker-sdk/client.rb#L393-L415.

It looks to me like it's avoid calling the block handler if the response code isn't 200, but it's not actually raising an exception. The `unless progress` thing is also always true if the HTTP connection opened, even if the response status is not 200, so we found in practice that our code ignored this error.

I'm guessing this should be refactored to use the error handling code here? https://github.com/looker-open-source/looker-sdk-ruby/blob/92cb759090d44f9fa8524a552508d8e6cbcdcb6e/lib/looker-sdk/error.rb#L33-L58

### PR DESCRIPTION
**Pull request by drstrangelooker**

### CODE REVIEW

File: lib/looker-sdk/client.rb [hunk 1]
```diff
@@ -396,14 +397,16 @@ def build_response(connection, request)
           http.read_timeout = connection.options.timeout rescue 60
 
           http.request(http_request) do |response|
+            if response.code != "200"
+              error = LookerSDK::Error.from_response(response)
```
Comment (jkaster) [inline lib/looker-sdk/client.rb:402]: this bypasses the network read error?

**Review by jkaster — APPROVED** [review]
Quick work!

File: lib/looker-sdk/client.rb (hunk 1 above)
Comment (drstrangelooker) [reply lib/looker-sdk/client.rb:402]: The original code looked for 200 and did something, but had no handling for a non-200 response. This raises an error on any non-200.

Comment (drstrangelooker) [reply lib/looker-sdk/client.rb:402]: I handled the network read error that @aschott-looker was having in https://github.com/looker-open-source/looker-sdk-ruby/pull/100 using the same code that he monkey patched.

File: lib/looker-sdk/client.rb [hunk 2]
```diff
@@ -396,14 +397,16 @@ def build_response(connection, request)
           http.read_timeout = connection.options.timeout rescue 60
 
           http.request(http_request) do |response|
+            if response.code != "200"
```
Comment (jkaster) [inline lib/looker-sdk/client.rb]: oh yeah, there are other valid non-200 error codes (2xx) that could be received from the API that aren't an error.

**Review by jkaster — CHANGES_REQUESTED** [review]
I think you need to expand allowing some other 2xx error codes?

File: lib/looker-sdk/client.rb (hunk 2 above)
Comment (drstrangelooker) [reply lib/looker-sdk/client.rb]: In the past we were only working with 200 responses. What other codes might we want?

Comment (drstrangelooker) [reply lib/looker-sdk/client.rb]: 200 OK
201 Created
202 Accepted
203 Non-Authoritative Information
204 No Content
205 Reset Content
206 Partial Content
207 Multi-Status
208 Already Reported
226 IM Used

**Review by jkaster — APPROVED** [review]

**Pull request merged**